### PR TITLE
fix/HIV app and Guppy

### PR DESCRIPTION
**Pull request by mfshao**
Let NIAID's HIV app support Guppy

Please make sure to follow the [DEV guidelines](https://gen3.org/resources/developer/dev-introduction/) before asking for review.

### New Features
- Replace Arranger by Guppy for HIV apps

### CODE REVIEW

File: src/GuppyDataExplorer/ExplorerFilter/index.jsx [hunk 1]
```diff
@@ -38,6 +38,10 @@ class ExplorerFilter extends React.Component {
         }
       }
     }
+    return null;
```
Comment (mfshao) [inline src/GuppyDataExplorer/ExplorerFilter/index.jsx]: to address a warning found in console while testing using dev.html.
`Warning: ExplorerFilter.getSnapshotBeforeUpdate(): A snapshot value (or null) must be returned. You have returned undefined.`

File: src/GuppyDataExplorer/ExplorerFilter/index.jsx [hunk 2]
```diff
@@ -38,6 +38,10 @@ class ExplorerFilter extends React.Component {
...
+  }
+
+  componentDidUpdate() {
```
Comment (mfshao) [inline src/GuppyDataExplorer/ExplorerFilter/index.jsx]: to address a warning found in console while testing using dev.html.
`Warning: ExplorerFilter: getSnapshotBeforeUpdate() should be used with componentDidUpdate(). This component defines getSnapshotBeforeUpdate() only.`

**Comment by mfshao** [comment]
deployed in https://qa-niaid.planx-pla.net

**Comment by mfshao** [comment]
> When I run this locally (qa-niaid) I get: SyntaxError: Unexpected token u in JSON at position 0
> 
> but maybe my local config is out of date? Is it deployed to qa-niaid where we can test there?

hmmm... yeah it is deployed on QA-NIAID now

File: src/HIVCohortFilter/HIVCohortFilterCase.jsx [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
   // Base class for the 3 NDH cohort filter apps. Meant to facilitate code reuse
-  static performQuery(queryString) {
+  static performQuery(queryString, variableString) {
+    const graphqlUrl = useGuppyForExplorer ? guppyGraphQLUrl : arrangerGraphqlPath;
     return fetchWithCreds({
-      path: `${arrangerGraphqlPath}`,
+      path: `${graphqlUrl}`,
       body: JSON.stringify({
         query: queryString,
+        variables: JSON.parse(variableString),
```
Comment (ZakirG) [inline src/HIVCohortFilter/HIVCohortFilterCase.jsx]: In the case where we use arranger, variableString will be undefined right? And I think that will lead to the error Uncaught SyntaxError: Unexpected token u in JSON at position 0

Comment (mfshao) [reply src/HIVCohortFilter/HIVCohortFilterCase.jsx]: nice catch! fixed in https://github.com/uc-cdis/data-portal/pull/558/commits/1dd79067b948a0cd77857bbc25b6548e37931ed9

**Pull request merged**